dlp/snippets/inspect/inspect_test.go

```diff
@@ -681,3 +685,115 @@ func TestInspectTableWithCustomHotword(t *testing.T) {
+
+	if err := InspectGCSFileSendToScc(&buf, tc.ProjectID, gcsPath); err != nil {
+		t.Fatal(err)
+	}
+
+	got := buf.String()
+	if want := "Job created successfully:"; !strings.Contains(got, want) {
+		t.Errorf("TestInspectGCSFileSendToScc got %q, want %q", got, want)
+	}
+}
+
+func filePathtoGCS(t *testing.T, projectID string) (string, error) {
+	t.Helper()
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return "", err
+	}
+	defer client.Close()
+	u := uuid.New().String()[:8]
+	bucketName := "dlp-go-lang-test" + u
+	dirPath := "update-stored-infoType-data/"
+
+	// Check if the bucket already exists.
+	bucketExists := false
+	_, err = client.Bucket(bucketName).Attrs(ctx)
+	if err == nil {
+		bucketExists = true
+	}
+
+	// If the bucket doesn't exist, create it.
+	if !bucketExists {
+		if err := client.Bucket(bucketName).Create(ctx, projectID, &storage.BucketAttrs{
+			StorageClass: "STANDARD",
+			Location:     "us-central1",
+		}); err != nil {
+			log.Fatalf("Failed to create bucket: %v", err)
+		}
+		fmt.Printf("Bucket '%s' created successfully.\n", bucketName)
+	} else {
+		fmt.Printf("Bucket '%s' already exists.\n", bucketName)
+	}
+
+	// Check if the directory already exists in the bucket.
+	dirExists := false
+	query := &storage.Query{Prefix: dirPath}
+	it := client.Bucket(bucketName).Objects(ctx, query)
+	_, err = it.Next()
+	if err == nil {
+		dirExists = true
+	}
+
+	// If the directory doesn't exist, create it.
+	if !dirExists {
+		obj := client.Bucket(bucketName).Object(dirPath)
+		if _, err := obj.NewWriter(ctx).Write([]byte("")); err != nil {
+			log.Fatalf("Failed to create directory: %v", err)
+		}
+		fmt.Printf("Directory '%s' created successfully in bucket '%s'.\n", dirPath, bucketName)
+	} else {
+		fmt.Printf("Directory '%s' already exists in bucket '%s'.\n", dirPath, bucketName)
+	}
+
+	// file upload code
+
+	// Open local file.
+	file, err := ioutil.ReadFile(filePathToUpload)
+	if err != nil {
+		log.Fatalf("Failed to read file: %v", err)
+	}
+
+	// Get a reference to the bucket
+	bucket := client.Bucket(bucketName)
+
+	// Upload the file
+	object := bucket.Object(inspectsGCSTestFileName)
+	writer := object.NewWriter(ctx)
+	_, err = writer.Write(file)
+	if err != nil {
+		log.Fatalf("Failed to write file: %v", err)
+	}
+	err = writer.Close()
+	if err != nil {
+		log.Fatalf("Failed to close writer: %v", err)
+	}
+	fmt.Printf("File uploaded successfully: %v\n", inspectsGCSTestFileName)
+
+	// Check if the file exists in the bucket
+	_, err = bucket.Object(inspectsGCSTestFileName).Attrs(ctx)
+	if err != nil {
+		if err == storage.ErrObjectNotExist {
+			fmt.Printf("File %v does not exist in bucket %v\n", inspectsGCSTestFileName, bucketName)
+		} else {
+			log.Fatalf("Failed to check file existence: %v", err)
+		}
+	} else {
+		fmt.Printf("File %v exists in bucket %v\n", inspectsGCSTestFileName, bucketName)
+	}
+
+	GCSUri := fmt.Sprint("gs://" + bucketName + "/" + dirPath + "test.txt")
```

Comment:
I am wondering when do these buckets get cleaned?

Reply:
Thank you for your review @annadata, In this implementation, I have not included a cleanup code explicitly. However, the code is designed to check if the bucket already exists before creating a new one. If the bucket already exists, the code will utilize the existing bucket instead of creating a duplicate.

If you prefer to clean up the bucket, I can add an additional function to handle that task. Please let me know which approach you would like me to follow.

Reply:
@AarshD-crest We should clean up the bucket created by the test. Please look into this.

Reply:
Updated it as per your suggestions @telpirion.

Reply:
comment: Let's try the `testutil.CreateBucket()` function here instead of rolling our own Storage bucket creation function.

Reply:
Thank You @telpirion for your comment. To set up the bucket using `testutil.CreateTestBucket()` the function expects a variable of type `*testing.T` which is not feasible to provide from TestMain(m *testing.M), Let me know how to proceed further,

Reply:
That's a good point. In that case, you can move the call to `testutil.CreateTestBucket()` into the body of the test. 

We (the repo maintainers) should investigate creating a new version of `CreateTestBucket()` that can execute in the context of `TestMain()`.